data/data-pipeline/data_pipeline/etl/sources/cdc_svi_index/etl.py

```diff
@@ -0,0 +1,114 @@
+            + field_names.PERCENTILE_FIELD_SUFFIX,  # SE prefix references "socioeconomic"
+            field_names.CDC_SVI_INDEX_HOUSEHOLD_THEME_COMPOSITION_FIELD
+            + field_names.PERCENTILE_FIELD_SUFFIX,
+            field_names.CDC_SVI_INDEX_LANGUAGE_THEME_FIELD
+            + field_names.PERCENTILE_FIELD_SUFFIX,
+            field_names.CDC_SVI_INDEX_HOUSING_TRANSPORTATION_FIELD
+            + field_names.PERCENTILE_FIELD_SUFFIX,
+            field_names.CDC_SVI_INDEX_RPL_THEMES_OVERALL_FIELD
+            + field_names.PERCENTILE_FIELD_SUFFIX,
+            # Derived columns
+            field_names.CDC_SVI_INDEX_THEMES_PRIORITY_COMMUNITY,
+        ]
+
+        self.df: pd.DataFrame
+
+    def extract(self) -> None:
+        logger.info("Extracting 43 MB CDC SVI INDEX")
+        self.df = pd.read_csv(
+            filepath_or_buffer=self.CDC_SVI_INDEX_URL,
+            dtype={self.CDC_SVI_INDEX_TRACTS_FIPS_CODE: "string"},
+            low_memory=False,
+        )
+
+    def transform(self) -> None:
+        logger.info("Starting CDC SVI INDEX transform")
+        # Note: In this dataset all US census tracts are ranked against one another.
+        # Puerto Rico is not included in this dataset
+        self.df.rename(
+            columns={
+                self.CDC_SVI_INDEX_TRACTS_FIPS_CODE: self.GEOID_TRACT_FIELD_NAME,
+                "RPL_THEME1": field_names.CDC_SVI_INDEX_SE_THEME_FIELD
+                + field_names.PERCENTILE_FIELD_SUFFIX,
+                "RPL_THEME2": field_names.CDC_SVI_INDEX_HOUSEHOLD_THEME_COMPOSITION_FIELD
+                + field_names.PERCENTILE_FIELD_SUFFIX,
+                "RPL_THEME3": field_names.CDC_SVI_INDEX_LANGUAGE_THEME_FIELD
+                + field_names.PERCENTILE_FIELD_SUFFIX,
+                "RPL_THEME4": field_names.CDC_SVI_INDEX_HOUSING_TRANSPORTATION_FIELD
+                + field_names.PERCENTILE_FIELD_SUFFIX,
+                "RPL_THEMES": field_names.CDC_SVI_INDEX_RPL_THEMES_OVERALL_FIELD
+                + field_names.PERCENTILE_FIELD_SUFFIX,
+            },
+            inplace=True,
+            errors="raise",
+        )
+
+        # From page 2 in the documentation:
+        # Tracts with zero estimates for total population (N = 645 for the U.S.) were removed during the ranking
+        # process. These tracts were added back to the SVI databases after ranking. The TOTPOP field value is 0,
+        # but the percentile ranking fields (RPL_THEME1, RPL_THEME2, RPL_THEME3, RPL_THEME4, and
+        # RPL_THEMES) were set to -999.
+        # ▪ For tracts with > 0 TOTPOP, a value of -999 in any field either means the value was unavailable from the
+        # original census data or we could not calculate a derived value because of unavailable census data.
+        # ▪ Any cells with a -999 were not used for further calculations. For example, total flags do not include fields
+        # with a -999 value.
+        self.df = self.df.replace(-999, np.nan)
```

Comment:
I know that I am imputing this across the board in the matrix. However, since we are selecting a subset of columns for actual output and analysis, this is not consequential. I can add that in the comments?

Reply:
Up to you, I'm fine either way. I think you can leave it as is but feel free to add more if you want.